draft-pda-protocol.md

```diff
@@ -257,52 +260,48 @@ proof showing that the encoded measurement is valid.
-   input and proof into two shares, one for the leader and another for a helper.
-   Rather than send each share to each aggregator directly, the client encrypts
-   each share under the helper's public key and sends the ciphertext to the
-   leader. The client repeats this procedure for each helper specified by the
-   leader.
-1. **Verify:** The leader initializes the input-validation protocol by sending
-   the encrypted shares to the aggregators. If the input is deemed valid, then
-   each aggregator stores its input share for processing later on.
-1. **Collect:** Finally, the collector interacts with the aggregators in order
-   to obtain the final output of the protocol.
+Each PA task consists of two sub-protocols, *upload* and *collect*, which are
+executed concurrently. Each sub-protocol consists of a sequence of HTTP requests
+made from one entity to another.
+
+~~~~
++-------------+ 1.      +-------------+
+|             +--------->             |
+|   Client    |         |   Leader    |
+|             |    +----+             |
++------+------+    | 2. +------^------+
+       | 1.        |           |
+       |           |           |
+       |           |           | 2.
++------v------+    |    +------+------+
+|             <----+    |             |
+|   Helper    |         |  Collector  |
+|             |         |             |
++-------------+         +-------------+
+~~~~
+{: #pa-topology title="Who makes requests to whom while executing a PA task."}
+
+1. **Upload:** Each client assembles its measurements into an input for the
+   given PA protocol. It generates a proof of its input's validity and splits
+   the input and proof into two shares, one for the leader and another for a
+   helper. Rather than send each share to each aggregator directly, the client
+   sends a request to the helper to learn its public key, encrypts its share
+   under this public key, and sends the ciphertext to the leader. The leader's
+   share and the helper's encrypted share are referred to collectively as the
+   client's *report*.
+2. **Collect:** The collector makes a request to the leader in order to obtain the
+   final output of the protocol. Before the output can be computed, the
+   aggregators (i.e, the leader and helpers) need to have verified and
+   aggregated a sufficient number of inputs. Depending on the PA protocol, it
+   may be possible for the aggregators to do so immediately when reports are
+   uploaded. (See {{prio}}.) However, in general it is necessary for them to
+   wait until (a) enough reports have been uploaded and (b) the collector has
+   made a request. (See {{hits}}.)
+
+[TODO: Say that the protocol involves multiple helpers. The downside of using
+secret sharing is that the protocol requires at least two servers to be online
+during the entire data aggregation process. To ameliorate this problem, we run
+the protocol in parallel with multiple helpers.]
```

Comment:
Can we move this to an issue and discuss separately? What does "we run the protocol in parallel with multiple helpers" actually mean? (Who is "we"?)

Reply:
We decided previously that multiple helpers would be used for resiliency. This has already been baked into the spec. This TODO is merely about motivating and describing this design choice. There are two issues that are impacted by this: #4 and #22.

Reply:
I'll raise as a separate issue as I no longer thing this is necessary to bake into the protocol this way.

Reply:
Done -> https://github.com/abetterinternet/prio-documents/issues/68